tests/common/dualtor/dual_tor_utils.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import pytest
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(scope='session')
+def tor_mux_intf(duthosts):
+    '''
+    Returns the server-facing interface on the ToR to be used for testing 
+    '''
+    # The same ports on both ToRs should be connected to the same PTF port
+    dut = duthosts[0]
+    return sorted(dut.get_vlan_intfs(), key=lambda intf: int(intf.replace('Ethernet', '')))[0]
+
+
+@pytest.fixture(scope='session')
+def ptf_server_intf(duthosts, tor_mux_intf, tbinfo):
+    '''
+    Returns the ToR-facing interface on the PTF to be used for testing
+
+    This should be connected to the interface returned by `tor_mux_intf`
+    '''
+    mg_facts = duthosts[0].get_extended_minigraph_facts(tbinfo)
+    ptf_port_index = mg_facts['minigraph_ptf_indices'][tor_mux_intf]
+    ptf_intf = 'eth{}'.format(ptf_port_index)
+
+    logger.info("Using PTF server interface {} for test".format(ptf_intf))
+    return ptf_intf
+
+
+@pytest.fixture(scope='session')
+def t1_upper_tor_intfs(upper_tor_host, tbinfo):
+    '''
+    Gets the PTF ports connected to the upper ToR for the first T1
+    '''
+    return get_t1_ptf_ports(upper_tor_host, tbinfo)
+
+
+@pytest.fixture(scope='session')
+def t1_lower_tor_intfs(lower_tor_host, tbinfo):
+    '''
+    Gets the PTF ports connected to the lower ToR for the first T1
+    '''
+    return get_t1_ptf_ports(lower_tor_host, tbinfo)
+
+
+@pytest.fixture(scope='session')
+def upper_tor_host(duthosts):
+    '''
+    Gets the host object for the upper ToR
+
+    Uses the convention that the first ToR alphabetically by hostname is the upper ToR
```

Comment:
Why not using the convention that the first ToR defined in the duts list of testbed.csv is the upper ToR?

Reply:
This is the existing convention that is used during minigraph generation